addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHttpMethodScanRule.java

```diff
@@ -642,4 +577,123 @@ private void testHttpMethod(String httpMethod) throws Exception {
+                        Constant.messages.getString(
+                                "ascanbeta.insecurehttpmethod.trace.exploitable.desc", method))
+                .setUri(URI)
+                .setOtherInfo(
+                        Constant.messages.getString(
+                                "ascanbeta.insecurehttpmethod.trace.exploitable.extrainfo",
+                                randomcookievalue))
+                .setEvidence(randomcookievalue)
+                .setAlertRef(getId() + "-3");
+    }
+
+    private AlertBuilder buildConnectResponseAlert(String thirdpartyHost, String response) {
+        return buildBaseAlert(HttpRequestHeader.CONNECT)
+                .setDescription(
+                        Constant.messages.getString(
+                                "ascanbeta.insecurehttpmethod.connect.exploitable.desc",
+                                HttpRequestHeader.CONNECT))
+                .setOtherInfo(
+                        Constant.messages.getString(
+                                "ascanbeta.insecurehttpmethod.connect.exploitable.extrainfo",
+                                thirdpartyHost))
+                .setEvidence(response)
+                .setAlertRef(getId() + "-4");
+    }
+
+    private AlertBuilder buildHttpMethodAlert(
+            int riskLevel,
+            String httpMethod,
+            String exploitableDesc,
+            String exploitableExtraInfo,
+            String evidence) {
+        return buildBaseAlert(httpMethod)
+                .setRisk(riskLevel)
+                .setDescription(exploitableDesc)
+                .setOtherInfo(exploitableExtraInfo)
+                .setEvidence(evidence)
+                .setAlertRef(getId() + "-5");
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return List.of(
+                buildDeleteMethodAlert("GET, POST, OPTIONS, HEAD, DELETE").build(),
+                buildInsecureHttpMethodAlert(
+                                Alert.RISK_MEDIUM,
+                                "PUT",
+                                "PUT",
+                                "GET, POST, HEAD, OPTIONS, PUT, DELETE",
+                                "PUT")
+                        .build(),
+                buildTraceCookieAlert(
+                                "aB3kL9zT2vQw8Xy1RmNp6sGh4Ud7Fc0ZjK9yX2mP",
+                                "TRACE",
+                                "http://example.com/")
+                        .build(),
+                buildConnectResponseAlert(
+                                "www.google.com",
+                                "HTTP/1.1 200 OK\n"
+                                        + "Date: Sat, 01 Nov 2025 19:33:10 GMT\n"
+                                        + "Content-Type: text/html; charset=ISO-8859-1\n"
+                                        + "Server: gws\n"
+                                        + "Connection: close")
+                        .build(),
+                buildHttpMethodAlert(
```

Comment:
This still needs to be extracted to its own method, it's also missing the webdav which IMO should have its own ref (i.e the one in line 569 should be replaced with two build/raise calls in lines 545 and 569, having all the data already filled in).